cmd/suseconnect/suseconnect.go

```diff
@@ -390,3 +401,37 @@ func fileExists(path string) bool {
 func isSumaManaged() bool {
 	return fileExists("/etc/sysconfig/rhn/systemid")
 }
+
+func processToken(token string) (string, error) {
+	var reader io.Reader
+
+	if strings.HasPrefix(token, "@") {
+		tokenFilePath := strings.TrimPrefix(token, "@")
+		file, err := os.Open(tokenFilePath)
+		if err != nil {
+			return "", fmt.Errorf("failed to open token file '%s': %w", tokenFilePath, err)
+		}
+		defer file.Close()
+		reader = file
+	} else if token == "-" {
+		reader = os.Stdin
+	} else {
+		return token, nil
+	}
+
+	return readTokenFromReader(reader)
+}
+
+func readTokenFromReader(reader io.Reader) (string, error) {
+	bufReader := bufio.NewReader(reader)
+	tokenBytes, err := bufReader.ReadString('\n')
+	if err != nil && err != io.EOF { // Handle potential errors, including EOF
```

Comment:
Don't inline comments, put them before the line instead. Hence:

```go
// Handle potential errors, including EOF
if err != nil && err != io.EOF {
```

Reply:
removed the inline comment. not required as such to put.